### ISSUE TITLE
Consider using `lodash`.

### ISSUE THREAD

**Issue by wtgtybhertgeghgtwtg** [lerna/lerna#553]
Currently, `lerna` depends on `lodash.find` and `lodash.unionwith`.  However, since it also depends on `inquirer`, `lodash` is already in the dependency tree, so `lodash.find` and `lodash.unionwith` are extra dependencies duplicating the functionality of a library that's already there.  Would you consider dropping `lodash.find` and `lodash.unionwith` and just depending on `lodash`?
It would also help simplify some things.  [For example](https://github.com/lerna/lerna/blob/aeb775d1fec9988f9165cbfad82db2d4fb796c7b/src/PackageGraph.js#L35-L45)
```
const depNames = Object.keys(dependencies);

for (let d = 0; d < depNames.length; d++) {
  const depName = depNames[d];
  const depVersion = dependencies[depName];
  const packageNode = this.nodesByName[depName];

  if (packageNode && semver.satisfies(packageNode.package.version, depVersion)) {
    node.dependencies.push(depName);
  }
}
```
can be rewritten as
```
import { forEach } from 'lodash';
forEach(dependencies, (depVersion, depName) => {
  const packageNode = this.nodesByName[depName];

  if (packageNode && semver.satisfies(packageNode.package.version, depVersion)) {
    node.dependencies.push(depName);
  }
});
```

**Comment by gigabo** [comment]
Oh, good idea @wtgtybhertgeghgtwtg.  If it's already a transitive dep seems like we may as well.

**Comment by noherczeg** [comment]
I believe lodash's forEach will be slower by a considerable amount. If the code shown is a performance critical piece I think it's not worth it.

**Comment by wtgtybhertgeghgtwtg** [comment]
For the example above, unless there are hundreds of dependencies (explicit, not transitive), any performance loss would have to be measured in microseconds.  I can't speak for anyone else, but I prefer simplicity over imperceivable performance.

**Comment by noherczeg** [comment]
Native forEach looks almost the same as the one in lodash yet they didn't use it when they wrote that code. Bringing in a util lib for a simple iteration increases coupling, and in the current case comes with a minuscule perf loss (which I agree should be barely noticeable).

The less dependencies a lib and it's modules have, the better (transitive doesn't count since third parties can remove them at any time, so it's not that great of a reason imho).

**Comment by wtgtybhertgeghgtwtg** [comment]
It's not just for iteration, `lodash.forEach` is just something that can be done if `lodash` is used.
I admit that it's not a strong argument, but it would still reduce the dependencies.

**Comment by doug-wade** [comment]
While generally I agree with @noherczeg that we should prefer JavaScript native constructs and functions, I find @wtgtybhertgeghgtwtg "it would reduce the dependencies" argument pretty compelling -- long install times can make CI builds particularly onerous.  There's also a fair number of utilities in lodash that aren't present as js constructs, like [flatten](https://lodash.com/docs/4.17.4#flatten) and [zip](https://lodash.com/docs/4.17.4#zip).  These may seem (or even be) trivial to implement, I'd still prefer to write fewer of these small helpful functions ourselves, and offload as much of that complexity to lodash as possible.  We'll just have to be vigilant in code reviews to make sure we're using native constructs when appropriate, and that seems reasonable to me (since the performance boosts are negligible, and we're already paying the price to install the dependency).  I think this is a home run @wtgtybhertgeghgtwtg.

**Comment by wtgtybhertgeghgtwtg** [comment]
Possible PR: #562.  Only replaces packages, does not make use of `lodash` outside of that.  That can be done later, I guess.

**Comment by wtgtybhertgeghgtwtg** [comment]
Closed with #562.  I can move some `Object.keys` stuff to `lodash` methods if you want, but that's probably too opinionated for me to do unless somebody asks.
